frame/election-provider-multi-phase/src/unsigned.rs

```diff
@@ -144,49 +144,63 @@ impl<T: Config> Pallet<T> {
 			Self::snapshot().ok_or(MinerError::SnapshotUnAvailable)?;
 		let desired_targets = Self::desired_targets().ok_or(MinerError::SnapshotUnAvailable)?;
 
-		// closures.
+		// now make some helper closures.
 		let cache = helpers::generate_voter_cache::<T>(&voters);
 		let voter_index = helpers::voter_index_fn::<T>(&cache);
 		let target_index = helpers::target_index_fn::<T>(&targets);
 		let voter_at = helpers::voter_at_fn::<T>(&voters);
 		let target_at = helpers::target_at_fn::<T>(&targets);
 		let stake_of = helpers::stake_of_fn::<T>(&voters, &cache);
 
-		let ElectionResult { assignments, winners } = election_result;
+		// Compute the size of a compact solution comprised of the selected arguments.
+		//
+		// This function completes in `O(edges)`; it's expensive, but linear.
+		let encoded_size_of = |assignments: &[IndexAssignmentOf<T>]| {
+			CompactOf::<T>::try_from(assignments).map(|compact| compact.encoded_size())
+		};
+
+		let ElectionResult { mut assignments, winners } = election_result;
+		// Sort the assignments by reversed voter stake. This ensures that we can efficiently truncate the list.
```

Comment:
Personal request: wrap lines (specially doc which is trivial) at 100 lines only. I always use https://marketplace.visualstudio.com/items?itemName=stkb.rewrap

Reply:
FYI the substrate standard is 100, only 120 is special circumstances.

Reply:
859555a51

Sorry, yes. I do try to keep to that limit, but I've got my editor set up to inject type annotations inline everywhere. Those annotations are extremely helpful, but they do mean that the apparent line length bears very little relation to the actual line length. I often just don't notice when the lines go overlong.

Reply:
yeah I mean these are soft rules I would say, but since I am pretty anal about this and the files that I touch are strictly 100, then if someone else edits them and doe snot respect this, look a bit strange. Thus, I named it as a personal request :D I hope the workflow works for you with that extension, else not a big deal.